lookout/style/typos/utils.py

```diff
@@ -0,0 +1,177 @@
+from itertools import chain
+
+from gensim.models.fasttext import FastText
+import numpy
+import pandas
+
+
+def collect_embeddings(fasttext: FastText, tokens) -> numpy.ndarray:
+    return numpy.array([fasttext.wv[token] for token in tokens])
+
+
+def read_frequencies(file: str) -> dict:
+    """
+    Read token frequencies from the file.
+    :param file: Path to the file containing tokens with frequencies.
+                Every line has format "token count"
+    :return: Dictionary of tokens frequencies
+    """
+    frequencies = {}
+    with open(file, "r") as f:
+        for line in f:
+            split = line.split()
+            frequencies[split[0]] = int(split[1])
+    return frequencies
+
+
+def read_vocabulary(file: str) -> list:
+    """
+    Read vocabulary tokens from the text file
+    :param file: Text file used to generate vocabulary of corrections candidates.
+            First token in every line split is added to the vocabulary
+    :return: List of tokens of the vocabulary
+    """
+    with open(file, "r") as f:
+        tokens = [line.split()[0] for line in f]
+    return tokens
+
+
+def flatten(data: pandas.DataFrame, column: str, new_column: str,
+            apply_function=lambda x: x) -> pandas.DataFrame:
+    """
+    Flatten dataframe on "column" with extracted elements put to "new_column"
+    """
+    flat_column = data[column].apply(apply_function).tolist()
+    flat_values = numpy.repeat(data.values,
+                               repeats=numpy.array(list(map(lambda x: len(x), flat_column))),
+                               axis=0)
+    flat_column = list(chain.from_iterable(flat_column))
+    result = pandas.DataFrame(flat_values, columns=data.columns)
+    result[new_column] = flat_column
+    return result.infer_objects()
+
+
+def flatten_data(data: pandas.DataFrame, new_column_name="identifier") -> pandas.DataFrame:
+    """
+    Flatten identifiers data in column "token_split"
+    :param data: DataFrame containing column "token_split" with splitted identifiers
+        either as strings or as lists of tokens
+    :param new_column_name: Name of column to put tokens from splits to
+    :return:
+    """
+    apply_function = (lambda x: x) if isinstance(data.token_split.tolist()[0], list) \
+        else (lambda x: str(x).split())
+    return flatten(data, "token_split", new_column_name, apply_function=apply_function)
+
+
+def add_context_info(data: pandas.DataFrame) -> pandas.DataFrame:
+    """
+    Split context of identifier on before and after part
+    :param data: DataFrame. Column "token_split" will be used for
+        creating context info if present
```

Comment:
Please align comments at the new line as in all other code - example https://github.com/src-d/style-analyzer/blob/master/lookout/style/format/features.py#L222

Reply:
and in other places too.